src/macros.rs

```diff
@@ -91,42 +57,17 @@ pub fn init_panic_handler() {
 #[macro_export]
 #[cfg(not(feature = "no-std"))]
 macro_rules! body {
-    ($gc:ident: $(())? $code:block) => {{
-        #[allow(unused_variables)]
+    ($gc:ident: $code:block) => {{
         let $gc = unsafe { $crate::Runtime::recover_handle() };
 
         // Ensure panic handler is initialized
         #[cfg(not(feature = "no-std"))]
-        $crate::init_panic_handler();
+        $crate::inital_setup();
 
-        #[allow(unused_mut)]
-        let mut r = |#[allow(unused_variables)] $gc: &mut $crate::Runtime| $code;
-        r($gc)
-    }};
-    ($gc:ident: ($($param:ident),+) $code:block) => {{
-        let $gc = unsafe { $crate::Runtime::recover_handle() };
-
-        // Ensure panic handler is initialized
-        #[cfg(not(feature = "no-std"))]
-        $crate::init_panic_handler();
-
-        struct __Values  {
-           $($param: $crate::Value),*
-        }
-
-        let values = __Values { $(
-            $param,
-        )*};
-
-        let ($($param),+) = $crate::interop::ocaml_frame!($gc, ($($param),+), {
-            ($(
-                unsafe { $crate::Value::new($param.keep_raw::<$crate::Value>(values.$param.0).get_raw()) }
-            ),+)
-        });
-
-        #[allow(unused_mut)]
-        let mut r = |$gc: &mut $crate::Runtime| $code;
-        r($gc)
+        {
+            let _ = &$gc;
```

Comment:
I was thinking, is it possible to make it so that the "unused variable" warning is silenced only when the user doesn't specify a name for the "gc" variable? So that if I don't specify the variable name because I will not use it, I don't get a warning, but if I specify a name and don't use it, I then do get the warning in that case.